### PR TITLE
feat: allow user to run ssh communicator through bastion

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
Allows users to use a bastion host to run ssh connections through. This feature is helpful if creating hosts in a publicly inaccessible network.

This enables us to run tests in vsphere for rhel 7.9 and 8.4

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test/infra/vsphere/packer-vsphere-airgap.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,4 @@
+packer:
+  ssh_bastion_username: "${SSH_BASTION_USERNAME}"
+  ssh_bastion_host: "${SSH_BASTION_HOST}"
```
Comment (supershal) [inline test/infra/vsphere/packer-vsphere-airgap.yaml.tmpl:3]: how are you extracting bastion ip from the terraform output?

File: test/infra/vsphere/export.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+# shellcheck disable=SC2001
+echo "${SSH_BASTION_KEY_CONTENTS}" | sed 's/\\n/\n/g' >> vsphere-tests.pem
```
Comment (supershal) [inline test/infra/vsphere/export.sh:3]: should we generate a temp keypair if CI does not provide `SSH_BASTION_KEY_CONTENTS` ?

**Review by supershal — CHANGES_REQUESTED** [review]
Thanks for making this very critical and challenging feature work.

File: test/infra/vsphere/packer-vsphere-airgap.yaml.tmpl [hunk 3]
```diff
@@ -0,0 +1,4 @@
+packer:
+  ssh_bastion_username: "${SSH_BASTION_USERNAME}"
```
[the remark this answers is not in the document]
Comment (faiq) [reply test/infra/vsphere/packer-vsphere-airgap.yaml.tmpl:2]: You got it dude.

**Review by supershal — APPROVED** [review]

**Pull request merged**